### ISSUE TITLE
Microsoft.CodeAnalysis.Common package 4.3.1 references invalid 4.3.0 assembly version for netstandard2.0

### ISSUE THREAD

**Issue by mnmr** [dotnet/roslyn#64926]
The Microsoft.CodeAnalysis.Common package version 4.3.1 contains a reference to a non-existing 4.3.0 version, which causes code referencing the 4.3.1 package to fail on build.

We have a source generator project (so it targets netstandard2.0) that references Microsoft.CodeAnalysis.Common. The solution has a Directory.Build.targets specifying the version of the assembly to use. When this is 4.2.0 the build succeeds everywhere. When updating to 4.3.1, the build fails when run on Azure DevOps. We likely have some locally cached package for 4.3.0 that makes it work locally; the DevOps build is clean every time.

The `deps.json` file also shows the version mismatch:

```
"Microsoft.CodeAnalysis.Common/4.3.1": {
        "dependencies": {
          "Microsoft.CodeAnalysis.Analyzers": "3.3.3",
          "System.Collections.Immutable": "6.0.0",
          "System.Memory": "4.5.4",
          "System.Reflection.Metadata": "5.0.0",
          "System.Runtime.CompilerServices.Unsafe": "6.0.0",
          "System.Text.Encoding.CodePages": "6.0.0",
          "System.Threading.Tasks.Extensions": "4.5.4"
        },
        "runtime": {
          "lib/netstandard2.0/Microsoft.CodeAnalysis.dll": {
            "assemblyVersion": "4.3.0.0",
            "fileVersion": "4.300.122.47918"
          }
        }

**Comment by jaredpar** [comment]
> The Microsoft.CodeAnalysis.Common package version 4.3.1 contains a reference to a non-existing 4.3.0 version, which causes code referencing the 4.3.1 package to fail on build.

The 4.3.0 version here is the assembly version. That does not change when we increment the patch version of the NuPkg. Looking at this break down that seems correct to me. There are a number of packages out which have successfully referenced this package. Do you have a minimal repro that demonstrates the problem you're seeing? 

@JoeRobich @jasonmalinowski

**Comment by jasonmalinowski** [comment]
Yeah, that matches my expectation for how we'd package this.

**Comment by mnmr** [comment]
> The 4.3.0 version here is the assembly version. That does not change when we increment the patch version of the NuPkg. Looking at this break down that seems correct to me. There are a number of packages out which have successfully referenced this package. Do you have a minimal repro that demonstrates the problem you're seeing?

I don't have a minimal repro at this time, however, the only thing changing between a successful build and a failed build is the version number of this package. The DevOps log file shows this error when attempting to build with a 4.3.1 reference:

```
  Restored /home/vsts/work/1/s/src/Common/RF.Contracts/RF.Contracts.csproj (in 2 ms).
  Restored /home/vsts/work/1/s/src/Common/RF.Generator.Serialization/RF.Generator.Serialization.csproj (in 2.8 sec).
  RF.Generator.Serialization -> /home/vsts/work/1/s/src/Common/RF.Generator.Serialization/bin/Release/netstandard2.0/RF.Generator.Serialization.dll
CSC : warning CS8032: An instance of analyzer RF.Generator.Serialization.JsonConverterSourceGenerator cannot be created from /home/vsts/work/1/s/src/Common/RF.Generator.Serialization/bin/Release/netstandard2.0/RF.Generator.Serialization.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/home/vsts/work/1/s/src/Common/RF.Contracts/RF.Contracts.csproj]
  RF.Contracts -> /home/vsts/work/1/s/src/Common/RF.Contracts/bin/Release/net6.0/RF.Contracts.dll
```

It could be an issue with the way packages are restored for a DevOps pipeline build, since it works locally. If there is nothing obviously wrong that can be remedied, then we'll likely just stay on 4.2.0 and hope the problem goes away when 4.4.0 is released (debugging DevOps build issues is no fun).

**Comment by jasonmalinowski** [comment]
Which SDK version are you building with in Azure DevOps there? That error would make me think you're still using the 4.2  compiler, which means you have a generator trying to use 4.3 but the compiler only has 4.2. You said you tried upgrading to 4.3.1 as a package which failed and you went back to 4.2, but does 4.3.0 also fail in the same way? There's supposed to be a more helpful error though rather than that generic failure; maybe we have a bug there.

If you have a binlog from that build that we could look at that could also confirm things.

**Comment by mnmr** [comment]
I am still investigating this issue, but it's been queued up behind some other issues I also have on my plate.. so this is just a keep-alive to let you know that more information is hopefully coming.

**Comment by mnmr** [comment]
I think the SDK may not have been updated in Azure DevOps at the time we tried this (as you suggested above), because now it works just fine. As such, I am closing this issue - thanks for your help!

That said, I am still somewhat perplexed as to why 4.3.0 has dissapeared from Nuget. I have the package locally but it can no longer be restored. No matter, toolchain is updated and 4.3.1 seems to work fine.
